Core/src/org/sleuthkit/autopsy/timeline/explorernodes/EventNode.java

```
@@ -155,10 +159,48 @@ public String getValue() throws IllegalAccessException, InvocationTargetExceptio
         }
 
         @Override
-        public void setValue(String t) throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+        final public void setValue(String t) throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
             String oldValue = getValue();
             value = t;
             firePropertyChange("time", oldValue, t); // NON-NLS
         }
     }
+
+    /**
+     * Factory method to create an EventNode form the event ID and the events
+     * model.
+     *
+     * @param eventID     The ID of the event this node is for.
+     * @param eventsModel The model that provides access to the events DB.
+     *
+     * @return An EventNode with the file (and artifact) backing this event in
+     *         its lookup.
+     */
+    public static EventNode createEventNode(final Long eventID, FilteredEventsModel eventsModel) {
+        /*
+         * Look up the event by id and creata an EventNode with the appropriate
+         * data in the lookup.
+         */
+        final SingleEvent eventById = eventsModel.getEventById(eventID);
+        try {
+            SleuthkitCase sleuthkitCase = Case.getCurrentCase().getSleuthkitCase();
+            AbstractFile file = sleuthkitCase.getAbstractFileById(eventById.getFileID());
+            if (file != null) {
+                if (eventById.getArtifactID().isPresent()) {
+                    BlackboardArtifact blackboardArtifact = sleuthkitCase.getBlackboardArtifact(eventById.getArtifactID().get());
+                    return new EventNode(eventById, file, blackboardArtifact);
+                } else {
+                    return new EventNode(eventById, file);
+                }
+            } else {
+                //This should never happen in normal operations
+                LOGGER.log(Level.WARNING, "Failed to lookup sleuthkit object backing TimeLineEvent."); // NON-NLS
```

Should throw instead of logging